tools/test/devicesim/scripts/run_targets.sh

```diff
@@ -1,56 +1,58 @@
 #!/bin/bash
 
+hostname=${HOSTNAME:-localhost}
```

Comment:
this breaks my docker-compose-linux mechanism a bit. i wanted to be able to pass in a hostname through the env parameter of a predictable hostname. With this I get a generated hostname and a message like:
```
Recreating devicesim_devicesim1_1 ... done
Starting devicesim_devicesim3_1   ... done
Starting devicesim_devicesim2_1   ... done
Attaching to devicesim_devicesim3_1, devicesim_devicesim2_1, devicesim_devicesim1_1
devicesim1_1  | Please add '172.25.0.11 dbf89d021135' to /etc/hosts and access with gNMI client at dbf89d021135:10161
devicesim2_1  | Please add '172.25.0.12 028b1358a44a' to /etc/hosts and access with gNMI client at 028b1358a44a:10161
devicesim3_1  | Please add '172.25.0.13 14bb9f76d9d0' to /etc/hosts and access with gNMI client at 14bb9f76d9d0:10161
```

It also breaks the docker-compose.yml - it doesn't seem to be catching the 'localhost' correctly and gives the oiutput below. The certs are not for localhost.
```
Recreating devicesim_devicesim1_1 ... done
Recreating devicesim_devicesim2_1 ... done
Recreating devicesim_devicesim3_1 ... done
Attaching to devicesim_devicesim2_1, devicesim_devicesim3_1, devicesim_devicesim1_1
devicesim2_1  | Please add '172.18.0.2 a6038da76cac' to /etc/hosts and access with gNMI/gNOI clients at a6038da76cac:10162:50002
devicesim1_1  | Please add '172.18.0.4 b66d6a578ccd' to /etc/hosts and access with gNMI/gNOI clients at b66d6a578ccd:10161:50001
devicesim2_1  | I0512 08:44:42.068867      21 gnoi_target.go:66] No credentials, setting Bootstrapping state.
devicesim1_1  | I0512 08:44:42.264359      21 gnoi_target.go:66] No credentials, setting Bootstrapping state.
devicesim3_1  | Please add '172.18.0.3 f5a338209297' to /etc/hosts and access with gNMI/gNOI clients at f5a338209297:10163:50003
devicesim1_1  | I0512 08:44:42.265982      21 gnoi_target.go:48] Starting gNOI server.
devicesim2_1  | I0512 08:44:42.070029      21 gnoi_target.go:48] Starting gNOI server.
devicesim3_1  | I0512 08:44:42.377225      21 gnoi_target.go:66] No credentials, setting Bootstrapping state.
devicesim3_1  | I0512 08:44:42.377909      21 gnoi_target.go:48] Starting gNOI server.
```

Reply:
We actually don’t have to change the Dockerfile at all. Well, I’ll still leave the permission change in there since that should not be done in the command. But this was actually a lazy hack due to a [bug/deficiency](https://github.com/kubernetes/kubernetes/issues/57726) in k8s. Kubernetes provides a `$HOSTNAME` environment variable, but the `$HOSTNAME` can’t be reassigned to another environment variable (though we should be able to). But technically we should either be using an init container to inject the service name into the container configuration or these options should be passed as arguments to the container where environment variables can be used. I will try one of the two.

Reply:
Sounds good